895357 - net-snmp-python adds zeros to end of IP address (IPADDR type), which is not valid

Source: upstream commit 234158b8e84cc204cbac96e6e9be6959635404b8

 --- a/python/netsnmp/client_intf.c	
+++ a/python/netsnmp/client_intf.c	
@@ -821,14 +821,14 @@ OCT:
 
       case TYPE_IPADDR:
         vars->type = ASN_IPADDRESS;
-        vars->val.integer = (long *)malloc(sizeof(long));
+        vars->val.integer = (in_addr_t *)malloc(sizeof(in_addr_t));
         if (val)
             *(vars->val.integer) = inet_addr(val);
         else {
             ret = FAILURE;
             *(vars->val.integer) = 0;
         }
-        vars->val_len = sizeof(long);
+        vars->val_len = sizeof(in_addr_t);
         break;
 
       case TYPE_OBJID:
